### DISCUSSION TITLE
Error with empty `skip-ci-flag` argument

### DISCUSSION THREAD

**Discussion by marinalimeira** [gruntwork-io/knowledge-base#232] [Help] [answered]
We're trying to use some of the ecs deploy runner set pieces to update our lambda versions by their s3 object tag. We're using very similar `update_args` as we'd find in our scripts that update our ecs applications:
```bash
# Invoke the ECS deploy runner using infrastructure-deployer to commit an update to the terragrunt configuration to
# deploy the newly built image. This commit will trigger the infrastructure CI/CD workflow to deploy the image.
function deploy_lambda_image {
  local -r lambda_version="$1"
  local -r deploy_path="$2"
  local -r lambda_tag_varname="$3"

  local -a update_args=(--aws-region "$REGION" --)
  update_args+=(terraform-applier terraform-update-variable)
  update_args+=(--git-branch-name "$DEFAULT_INFRA_LIVE_BRANCH")
  update_args+=(--vars-path "$deploy_path/terragrunt.hcl")
  update_args+=(--name "$lambda_tag_varname" --value "\"$lambda_version\"")
  update_args+=(--skip-ci-flag "")
  (eval infrastructure-deployer "${update_args[@]}")
}
```
Something is odd with `skip-ci-flag` I believe. We get this error when the value is empty string:
```ERROR: IndexError: pop from an empty deque```
Once we add some arbitrary string, it works, but in our other scripts, that field is not set. can anyone shed light on what we're missing? Thanks!
The error is coming from the python ecs runner invoker lambda.
Secondly, when we get past the skip-ci-flag error, it seems that the quotes around our lambda s3 object version are being stripped off. We're investigating whether this could be a script issue on our side...

**Comment by marinalimeira** [comment]
From what I can see in [the `terraform-update-variable` code](https://github.com/gruntwork-io/terraform-aws-ci/blob/master/modules/terraform-helpers/bin/terraform-update-variable#L15), the default value for `--skip-ci-flag` is `"[ci skip]"` so it's a odd that you need to set it to at least an empty string. 

> it seems that the quotes around our lambda s3 object version are being stripped off.

Do you have more details on this?

**Comment by marinalimeira** [comment, marked as answer]
The `[ci skip]` is a feature [from CircleCI to skip the build](https://circleci.com/docs/2.0/skip-build/). Is that what you what? Do you need to skip the build or not?

**Comment by dev-admin-gridline** [comment]
@marinalimeira sorry for the late reply, been a few crazy days over here :) Unforunately, I don't have much more details on hand I can provide, but if there's anything specific I can provide, I'd be happy to do so! 

Most importantly, we were able to find a workaround by just providing an arbitrary string for that argument. We do not want to skip-ci in this case. What is still odd to me is, in our ecs service we set up, we don't provide a value for the argument skip-ci-flag, we do not want to skip ci in that case, and we do not skip ci in that place. So, to your point, it is odd that the default value, which would skip the ci, is not doing so with an empty value.

secondly, on the issue of quotations being stripped off, that was an issue with our script we wrote, and that has been resolved.

Thanks for you help, but at this time there's nothing needed urgently from our side. would still love to understand the discrepancy I pointed out above, if possible.

Brad
